### ISSUE TITLE
Swagger2::Client: systematically croak() if $tx->error()

### ISSUE THREAD

**Issue by le-garff-yoann** [jhthorsen/swagger2#67]
Is it possible (with a blocking request) to return the `$tx->res()` (or `$tx`) object without `croak() if $tx->error()`(line n°84) ? Or to add an option to do this without breaking the current implementation ?

**Comment by jhthorsen** [comment]
The module is marked as "DRAFT", so I think any change can happen...

When that is said: It needs to be something which can be understandable in string context. We might have to implement an exception object for this to work. Or maybe we could have $client->tx to retrieve the last transaction? You would have to capture the failure in an ```eval{}``` though.

**Comment by le-garff-yoann** [comment]
Look nice (although the `eval` seem redundant to me).

The solution could also be something like that (?):

```perl
$ua->list_pets(
    parameters => {},
    sync_cb => sub { # "async_cb" key for async request
        my $tx = shift;

        $tx->res;
    } # if ref ne 'CODE'/key not specified, switch to the actual/default behavior ("understandable in string context" + sync request)
); # return with the callback if "async_cb" key is not specified
```

**Comment by jhthorsen** [comment]
Yeah... This is getting a bit dirty.

What do you think about 0ff7873?

**Comment by le-garff-yoann** [comment]
Very nice :+1: 

I'm ok with this.

**Comment by jhthorsen** [comment]
Cool :+1: 0.69 should be available soon for download from CPAN.

**Comment by jhthorsen** [comment]
Fixed in 0.69
